lib/llm/src/preprocessor/media/common.rs

```diff
@@ -0,0 +1,147 @@
+// SPDX-FileCopyrightText: Copyright (c) 2024-2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use base64::{Engine as _, engine::general_purpose};
+
+// Raw encoded media data (.png, .mp4, ...), optionally b64-encoded
+#[derive(Debug)]
+pub struct EncodedMediaData {
+    pub(crate) bytes: Vec<u8>,
+    pub(crate) b64_encoded: bool,
+}
+
+impl EncodedMediaData {
+    // Handles both web URLs (will download the bytes) and data URLs (will keep b64-encoded)
+    // This function is kept in tokio runtime so we do not want any expensive operations
```

Comment:
>     // This function is kept in tokio runtime so we do not want any expensive operations

I don't think we need that last comment. The `async` shows it's called from a runtime.

Reply:
Right, removed